src/ws/login.js

```diff
@@ -237,11 +237,13 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
         if (authorized.indexOf("password") !== -1)
             id("authorized-input").checked = true;
 
-        var os_release = JSON.stringify(environment["os-release"]);
+        var os_release = environment["os-release"];
+        if (os_release)
+            window.localStorage.setItem('os-release', JSON.stringify(os_release));
```

Comment:
Does that actually make any difference? In JavaScript (unlike in Python), an empty dict is considered true. `environment["os-release"]` by itself is always set in cockpithandlers.c `build_environment()` AFAICS.

Reply:
Yes this makes a difference. Old code put the string "undefined" in localStorage.

Reply:
Oh, so it will be `undefined`, not an empty array, I see. I didn't see how it got from the above `base_index.js` to here.